plugin/src/main/java/org/jenkinsci/plugins/workflow/cps/CpsFlowExecution.java

```diff
@@ -642,21 +648,324 @@ Invoker createInvoker() {
+            //   FAILED to parse 'stepName' due to...
+            overflowedClassNameReport = "presumed JSL step '" + overflowedClassName + "'";
+        }
+
+        // Short message suffices, not much that a pipeline developer
+        // can do with the stack trace into the guts of groovy
+        StringBuilder actionableMsg = new StringBuilder();
+        actionableMsg
+                .append("FAILED to parse ")
+                .append(overflowedClassNameReport)
+                .append(" due to MethodTooLargeException");
+        if (ecCount > 1) {
+            actionableMsg.append(" (and other issues)");
+        }
+        actionableMsg.append("; please refactor to simplify code structure");
+        if (overflowedClassNameReport.contains("WorkflowScript") || CLASSNAME_SCRIPTNUM_PATTERN.matcher(overflowedClassName).find())
+            actionableMsg.append(" and/or move logic to a Jenkins Shared Library");
+        if (xMsgStart.length() > 0) {
+            actionableMsg
+                    .append(":\n-----\n")
+                    .append(xMsgStart.toString());
+        }
+        if (overflowedClassNameBreadcrumbs.length() > 0) {
+            actionableMsg
+                    .append("\nGroovy code trail (mentions of pipeline WorkflowScript and/or your JSL in larger stack trace):\n")
+                    .append(overflowedClassNameBreadcrumbs);
+        }
+        if (xMsgStart.length() > 0) {
+            if (!(actionableMsg.substring(actionableMsg.length() - 1).equals("\n")))
+                actionableMsg.append("\n");
+            actionableMsg.append("-----");
+        }
+
+        // Make a full note in server log
+        METHOD_TOO_LARGE_LOGGER.log(Level.FINER,
+                "CpsFlowExecution.reportSuspectedMethodTooLarge: full original Throwable message:\n"
+                        + xStr);
+
+        if (ecCount > 1) {
+            // Not squashing with explicit MethodTooLargeException
+            // re-thrown below, in this codepath we have other errors.
+            return new RuntimeException(actionableMsg.toString(), x);
+        }
+
+        // ecCount == 1 exactly, this is the only problem we saw.
+        // Do not confuse pipeline devs by a wall of text in the
+        // build console, but let the full context be found in
+        // server log with some dedication. Note it is seen at
+        // a different logging verbosity level.
+        METHOD_TOO_LARGE_LOGGER.log(Level.FINE,
+                "CpsFlowExecution.reportSuspectedMethodTooLarge: detected details of MethodTooLargeException:\n"
+                        + mtlEx.getMessage());
+
+        actionableMsg
+                .append("\nComplete details can be seen in server log at FINE/FINER level ")
+                .append("(Jenkins admin access for ")
+                .append(METHOD_TOO_LARGE_LOGGER.getName())
+                .append(" is required)");
+
+        //return new RuntimeException(actionableMsg.toString(), mtlEx);
```

Comment:
Do not leave commented-out code in a final PR (unless there is a real possibility it should be resurrected some day soonish and a preceding comment explains those circumstances).